src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetKeyValueDropdown.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.ui.stepsdesign;
+
+import com.google.gson.JsonObject;
+import io.cdap.cdap.ui.utils.Commands;
+import io.cdap.cdap.ui.utils.Helper;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.WaitHelper;
+import io.cucumber.java.en.Then;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class WidgetKeyValueDropdown {
+
+  private String getCssSelectorForWidgetRow(String rowIndex) {
+    return Helper.getCssSelectorByDataTestId("convert") + " " + Helper.getCssSelectorByDataTestId(rowIndex);
+  }
+
+  private String getCssSelectorForKeyInWidgetRow(String rowIndex) {
+    return getCssSelectorForWidgetRow(rowIndex) + " " + Helper.getCssSelectorByDataTestId("key") + " input";
+  }
+
+  private String getCssSelectorForTypeValueInWidgetRow(String rowIndex) {
+    return getCssSelectorForWidgetRow(rowIndex) + " " + Helper.getCssSelectorByDataTestId("value") + " input";
```

Comment:
it could be `getCssSelectorForValueDropDownInWidgetRow(String rowIndex) + " input"`

Reply:
yes, Changed . Also, using common functions from helper class now.